host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs

```diff
@@ -50,14 +50,21 @@ private async Task Process(StreamingMessage msg)
                     Logger.LogTrace("Specialization request received.");
 
                     var envReloadRequest = msg.FunctionEnvironmentReloadRequest;
+                    var applicationExePath = PathUtils.GetApplicationExePath(envReloadRequest.FunctionAppDirectory);
+                    Logger.LogTrace($"application path {applicationExePath}");
+
+                    if (string.IsNullOrWhiteSpace(applicationExePath))
+                    {
+                        var ex = new InvalidOperationException($"Unable to find a valid function app payload at '{envReloadRequest.FunctionAppDirectory}'");
+                        responseMessage.FunctionEnvironmentReloadResponse = BuildFailedEnvironmentReloadResponse(ex);
```

Comment:
I'll admit I don't know how the host handles this. I assume it logs the exception and kills the host? Or does it retry?

Reply:
We log an error and set call the `SetException` method on TCS.
https://github.com/Azure/azure-functions-host/blob/c85df40731c1862d3b7b79fc0537044b5426c247/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L377C46-L377C46
The grpc channel to worker continue to stay open.

Reply:
This error will show up in the customer's App Insights, right?

Reply:
No. This does not go to customer logs.

Reply:
> We log an error and set call the `SetException` method on TCS. https://github.com/Azure/azure-functions-host/blob/c85df40731c1862d3b7b79fc0537044b5426c247/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L377C46-L377C46 The grpc channel to worker continue to stay open.

Found out that host code is not gracefully handling this case. Let me follow up tomorrow.

Reply:
I will be making a change in the host which will check whether we have a valid payload before initiating the language worker specialization.

The change in this PR is still valid because without this change, if the worker config file does not exist in the payload, we used to crash the process. with this change, it will not crash, but sends an env reload response with "Faiulre" as status. Host will shutdown the application in that case.